server/models/secret.js

```diff
@@ -0,0 +1,51 @@
+const { init } = require('../initdb');
+
+class Secret {
+    constructor(data){
+        this.id = data.id;
+        this.title = data.title;
+        this.pseudo = data.pseudo;
+        this.abstract = data.abstract;
+    }
+
+    static get all() {
+        return new Promise(async (resolve, reject) => {
+            try {
+                const db = await init();
+                const dbData = await db.collection('secrets').find({}).toArray()
+                const secrets = dbData.map(d => new Secret(d))
+                if(!secrets.length) {throw new Error('Dunno, no secrets here')}
+                resolve(secrets);
+            } catch (err) {
+                reject(`Error retrieving secrets: ${err.message}`)
+            }
+        })
+    }
+
+    static create(data){
+        return new Promise(async (resolve, reject) => {
+            try {
+                console.log("Create function - data: ", data)
+                const db = await init()
+                let secret = await db.collection('secrets').insertOne(data)
```

Comment:
mongodb is so much easier to work with for small db like we are doing here. Good choice